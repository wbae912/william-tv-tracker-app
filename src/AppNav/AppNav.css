.appnav-header {
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  background-color: rgb(78, 66, 61);
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 10;
  height: 70px;
}

.appnav-logo-header {
  display: flex;
  align-items: center;
  margin-left: 2%;
  width: 50%;
}

.fa-tv {
  cursor: pointer;
  display: inline-block;
  margin-right: 2%;
  color: black;
}

.appnav-logo-div {
  margin-right: 2%;
}

.appnav-title {
  display: inline-block;
  cursor: pointer;
  color: black;
  font-family: 'Esteban', serif;
  color: white;
  text-decoration: none;
  font-size: 20px;
}

.appnav {
  display: none;
}

.appnav-links {
  list-style: none;
}

.appnav-links li {
  display: inline-block;
  padding: 0 20px;
  transition: all 0.3s ease 0s;
  font-weight: bold;
  color: white;
  font-size: 12px;
  /* CHANGE FONT SIZE WHEN LARGER SCREEN */
}

.appnav-links li:hover {
  color: rgb(133, 70, 133);
}

.logout-button {
  background-color: transparent;
  color: white;
  border-radius: 10px;
  border: 1.5px solid white;
  width: 80px;
  height: 30px;
  font-size: 14px;
  transition-duration: 0.4s;
}

.logout-button:hover {
  background-color: white;
  color: black;
}

#dropdown-li {
  display: none;
}

/* HAMBURGER MENU STYLING */
.hamburger {
  color: white;
  height: 70px;
  line-height: 70px;
  width: 160px;
}

.hamburger-logo {
  font-weight: bold;
  font-size: 18px;
}

.menu {
  margin: 0;
  display: none;
  text-align: center;
}

.menu li {
  clear: right;
  display: block;
  border-bottom: 1px solid white;
  background-color: rgb(78, 66, 61);
  color: white;
  text-decoration: none;
  width: 160px;
  margin: 0;
  float: right;
  font-size: 14px;
}

.appnav-li-first {
  border-top: 1px solid white;
}

.hamburger-logo {
  cursor: pointer;
  float: right;
  margin-right: 10%
}

#toggle {
  display: none;
}

#toggle:checked + .menu {
  display: block;
}

.hamburger-logo:hover {
  color: rgb(133, 70, 133);
}
/* END HAMBURGER MENU STYLING */

@media only screen and (min-width: 1300px) {
  .hamburger {
    display: none;
  }

  /* DROPDOWN MENU STYLING */
  #dropdown-li {
    display: inline-block;
    height: 70px;
    line-height: 70px;
  }

  .dropbtn {
    background-color: rgb(78, 66, 61);
    color: white;
    font-family: 'Montserrat', sans-serif;
    padding: 0px 10px;
    margin-bottom: 0px;
    font-size: 16px;
    font-weight: bold;
    border: none;
    cursor: pointer;
  }

  .dropdown {
    position: relative;
    display: inline-block;
  }

  .dropdown-content {
    display: none;
    position: absolute;
    background-color: rgb(78, 66, 61);
    min-width: 160px;
    z-index: 1;
  }

  .dropdown-content a {
    color: white;
    padding: 12px 16px;
    text-decoration: none;
    display: block;
  }

  .dropdown-content a:hover {
    color: rgb(133, 70, 133);
  }

  .dropdown:hover .dropdown-content {
    display: block;
  }

  .dropdown:hover .dropbtn {
    color: rgb(133, 70, 133);
  }

  .dropdown-span {
    font-size: 13px;
  }
  /* END DROPDOWN MENU STYLING */

  .appnav {
    display: inline-block;
  }

  .appnav-links li {
    font-size: 16px;
    padding: 0 10px;
  }


}