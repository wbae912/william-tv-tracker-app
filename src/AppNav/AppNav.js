import React, { Component } from 'react';
import { Link, NavLink } from 'react-router-dom';
import TokenService from '../services/token-service';
import tvContext from '../Context';
import './AppNav.css';

export default class AppNav extends Component {

  static contextType = tvContext;

  handleLogoutClick = () => {
    TokenService.clearAuthToken();
  }

  render() {
    return (
      <header className="appnav-header">
        <div className="appnav-logo-header">
          <div className="appnav-logo-div">
            <Link to='/dashboard'><i className="fa fa-tv fa-2x"></i></Link>
          </div>
          <Link to='/dashboard'><h2 className="appnav-title">TV Tracker</h2></Link>
        </div>

        <nav className="appnav">
          <ul className="appnav-links">
            <NavLink to='/dashboard' activeClassName="selected" className="appnav-li"><li>Dashboard</li></NavLink>
            <li className="appnav-li" id="dropdown-li">
              <div className="dropdown">
                <button className="dropbtn">TV Show Status</button>
                <div className="dropdown-content">
                  <NavLink to='/plan-to-watch' activeClassName="selected" className="dropdown-link"><span className="dropdown-span">Plan to Watch</span></NavLink>
                  <NavLink to='/currently-watching' activeClassName="selected" className="dropdown-link"><span className="dropdown-span">Currently Watching</span></NavLink>
                  <NavLink to='/completed' activeClassName="selected" className="dropdown-link"><span className="dropdown-span">Completed</span></NavLink>
                </div>
              </div>
            </li>
            <NavLink to='/add-entry' activeClassName="selected" className="appnav-li"><li >Add New Show</li></NavLink>
            <Link to='/'>
              <li className="appnav-li">
                <button type="button" className="logout-button"
                  onClick={() => {this.handleLogoutClick(); this.context.toggleIsLoggedOff()}}>
                  Logout
                </button>
              </li>
            </Link>
          </ul>
        </nav>

        <div className="hamburger">
            <label htmlFor="toggle" className="hamburger-logo">&#9776;</label>
            <input type="checkbox" id="toggle"/>
            <ul className="menu">
                <Link to='/dashboard'><li className="appnav-li appnav-li-first">Dashboard</li></Link>
                <Link to='/plan-to-watch'><li className="appnav-li">Plan to Watch</li></Link>
                <Link to='/currently-watching'><li className="appnav-li">Currently Watching</li></Link>
                <Link to='/completed'><li className="appnav-li">Completed</li></Link>
                <Link to='/add-entry'><li className="appnav-li">Add New Show</li></Link>
                <Link to='/'>
                  <li className="appnav-li" 
                    onClick={() => {this.handleLogoutClick(); this.context.toggleIsLoggedOff()}}>
                    Logout
                  </li>
                </Link>
            </ul>
        </div>
        
      </header>
    )
  }
}