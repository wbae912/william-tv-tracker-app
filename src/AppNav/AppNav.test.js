import React from 'react';
import ReactDOM from 'react-dom';
import AppNav from './AppNav';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';

describe('AppNav component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter><AppNav /></MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><AppNav /></MemoryRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
  })
})