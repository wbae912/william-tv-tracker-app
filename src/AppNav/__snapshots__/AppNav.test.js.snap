// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNav component renders UI as expected 1`] = `
<header
  className="appnav-header"
>
  <div
    className="appnav-logo-header"
  >
    <div
      className="appnav-logo-div"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="fa fa-tv fa-2x"
        />
      </a>
    </div>
    <a
      href="/"
      onClick={[Function]}
    >
      <h2
        className="appnav-title"
      >
        TV Tracker
      </h2>
    </a>
  </div>
  <nav
    className="appnav"
  >
    <ul
      className="appnav-links"
    >
      <a
        aria-current={null}
        className="appnav-li"
        href="/dashboard"
        onClick={[Function]}
      >
        <li>
          Dashboard
        </li>
      </a>
      <li
        className="appnav-li"
        id="dropdown-li"
      >
        <div
          className="dropdown"
        >
          <button
            className="dropbtn"
          >
            TV Show Status
          </button>
          <div
            className="dropdown-content"
          >
            <a
              aria-current={null}
              className="dropdown-link"
              href="/plan-to-watch"
              onClick={[Function]}
            >
              <span
                className="dropdown-span"
              >
                Plan to Watch
              </span>
            </a>
            <a
              aria-current={null}
              className="dropdown-link"
              href="/currently-watching"
              onClick={[Function]}
            >
              <span
                className="dropdown-span"
              >
                Currently Watching
              </span>
            </a>
            <a
              aria-current={null}
              className="dropdown-link"
              href="/completed"
              onClick={[Function]}
            >
              <span
                className="dropdown-span"
              >
                Completed
              </span>
            </a>
          </div>
        </div>
      </li>
      <a
        aria-current={null}
        className="appnav-li"
        href="/add-entry"
        onClick={[Function]}
      >
        <li>
          Add New Show
        </li>
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <li
          className="appnav-li"
        >
          <button
            className="logout-button"
            onClick={[Function]}
            type="button"
          >
            Logout
          </button>
        </li>
      </a>
    </ul>
  </nav>
  <div
    className="hamburger"
  >
    <label
      className="hamburger-logo"
      htmlFor="toggle"
    >
      â˜°
    </label>
    <input
      id="toggle"
      type="checkbox"
    />
    <ul
      className="menu"
    >
      <a
        href="/dashboard"
        onClick={[Function]}
      >
        <li
          className="appnav-li appnav-li-first"
        >
          Dashboard
        </li>
      </a>
      <a
        href="/plan-to-watch"
        onClick={[Function]}
      >
        <li
          className="appnav-li"
        >
          Plan to Watch
        </li>
      </a>
      <a
        href="/currently-watching"
        onClick={[Function]}
      >
        <li
          className="appnav-li"
        >
          Currently Watching
        </li>
      </a>
      <a
        href="/completed"
        onClick={[Function]}
      >
        <li
          className="appnav-li"
        >
          Completed
        </li>
      </a>
      <a
        href="/add-entry"
        onClick={[Function]}
      >
        <li
          className="appnav-li"
        >
          Add New Show
        </li>
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <li
          className="appnav-li"
          onClick={[Function]}
        >
          Logout
        </li>
      </a>
    </ul>
  </div>
</header>
`;
